### PR TITLE
Add weather widget examples

### PR DESCRIPTION
**Pull request by Layerex**
Add a widget, which shows current state of weather in a specified location. That is my first pull request, I hope I did everything alright

### CODE REVIEW

**Comment by shdown** [comment]
1. Thanks for the pull request! I would like to merge it after the following issues are addressed.

2. I think it's better to use HTTPS, even if the particular content (weather data) is not security-sensitive — in order not to “set a bad example” for others. Also, you never know which content *will*  turn out to be security-sensitive in the future.

3. Both cURL and luacurl seem to be abandoned and unmaintained. Unfortunately, this situation is too common with Lua modules. I would rather use luasec and LuaSocket, which seem to be at least somewhat actively maintained. See the `examples/btc-price.lua` for how to perform an HTTPS request with it.

4. URL encoding is no hard task; I think it’s better to implement it manually rather than introduce a dependency:
```lua
function urlencode(s)
    return string.gsub(s, '[^-_.~a-zA-Z0-9]', function(c)
        return string.format('%%%02X', string.byte(c))
    end)
end
```

File: examples/dwm/weather.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+local location = ''
+local PERIOD = 900
+
+function get_weather(format)
+    -- encoding is needed to allow usage of special use characters
+    format = encoder.encode(format)
+    curl.easy{
+        url = string.format('%s.%s/%s?format=%s', lang, base_url, location, format),
+        writefunction = function (str)
+            res = str:gsub("\n","")
```
Comment (shdown) [inline examples/dwm/weather.lua:18]: Did you really intend to assign to global variable `res` here?

Comment (Layerex) [reply examples/dwm/weather.lua:19]: Fixed

File: examples/dwm/weather.lua [hunk 2]
```diff
@@ -0,0 +1,68 @@
+-- you need to install 'lua-curl' and 'gonapps-url-encoder' modules (e.g. with luarocks)
```
Comment (shdown) [inline examples/dwm/weather.lua:1]: You don’t need to install those anymore — the only dependency now is LuaSec :)

File: examples/dwm/weather.lua [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    else
+        return nil
+    end
+end
+
+widget = {
+    plugin = 'timer',
+    opts = {period = PERIOD},
+    cb = function()
+        text = get_weather('%l: %C %t(%f)')
```
Comment (shdown) [inline examples/dwm/weather.lua:61]: This assigns to a global variable.

**Comment by shdown** [comment]
Merged, thanks!

**Comment by NigamanRPG** [comment]
nice widget and contests interesting!

**Pull request merged**